x-pack/plugins/files/server/routes/file_kind/upload.ts

```diff
@@ -56,6 +62,12 @@ export const handler: FileKindsRequestHandler<Params, unknown, Body> = async (
       return res.badRequest({ body: { message: e.message } });
     } else if (e instanceof fileErrors.AbortedUploadError) {
       fileService.logger.error(e);
+      if (req.query.selfDestructOnAbort) {
```

Comment:
I think this should not be the default behaviour, because it could lead to a surprising situation for clients of the file service who have a failed upload and now their file SO also vanished.